core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -46,14 +49,41 @@
 import io.vertx.ext.web.Router;
 import io.vertx.ext.web.api.contract.openapi3.OpenAPI3RouterFactory;
 import io.vertx.ext.web.impl.BlockingHandlerDecorator;
+import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.jdbi.v3.core.Jdbi;
 
 public class Eth2Runner extends Runner {
   final Optional<SlashingProtection> slashingProtection;
 
-  public Eth2Runner(final Config config, final Optional<SlashingProtection> slashingProtection) {
+  public Eth2Runner(
+      final Config config,
+      final boolean slashingProtectionEnabled,
+      final String slashingProtectionDbUrl,
+      final String slashingProtectionDbUser,
+      final String slashingProtectionDbPassword) {
     super(config);
-    this.slashingProtection = slashingProtection;
+    this.slashingProtection =
+        createSlashingProtection(
+            slashingProtectionEnabled,
```

Comment:
can someone disable slashing protection, but still supply all fields? (yes, this one is my bugbear)

Reply:
yes they could. not sure what the answer is here. we could display a warning if any of these values are set and slashing protection isn't enabled?

Reply:
It's extremely useful to be able to have a config file with lots of things specified and then just change an `enabled: true` to `enabled: false` rather than having to delete any other relevant options.  Similarly it's very common to have a long CLI command and then just go back and edit one flag - very annoying and user hostile when it then errors and demands you delete flags that could have easily been ignored.

Reply:
changed to only validate the slashing db url if slashing is enabled. this is required if it is enabled to so we can't ignore this option otherwise web3signer will just fail to startup. if slashing isn't enabled we just ignore the options.